### DISCUSSION TITLE
Export Response from external library

### DISCUSSION THREAD

**Discussion by tihobanov** [SamWSoftware/serverless-auto-swagger#90] [Q&A]
I am using an external payments library and want to add the response to swagger docs, is there a way?

```
- external-library.d.ts
export declare class PaymentMethodsResponse {
    /**
    * Detailed list of payment methods required to generate payment forms.
    */
    'paymentMethods'?: Array<PaymentMethod>;
    /**
    * List of all stored payment methods.
    */
    'storedPaymentMethods'?: Array<StoredPaymentMethod>;
}

- types.d.ts
import {PaymentMethodsResponse} from 'external-library'

export type Response = PaymentMethodsResponse;

result: "string"

```

**Comment by bfaulk96** [comment]
No, all types that you want to surface in the Swagger docs must be defined in the file, they cannot be imported.  I believe it's a limitation with the `typeconv` library, which is what is used internally to generate the JSON schema. I know this quite inconvenient as it leads to duplicate declarations; it's a pain point for me as well. Hopefully we can figure out how to improve this in the future.
